src/Clients/Rest.php

```diff
@@ -57,10 +57,19 @@ protected function request(
         );
     }
 
-    private function getRestPath(string $path): string
+    protected function getRequestPath(string $path): string
     {
+        $path = parent::getRequestPath($path);
+        $path = preg_replace("/\.json$/", "", $path) . ".json";
+
         $apiVersion = Context::$API_VERSION;
-        return "admin/api/$apiVersion/$path.json";
+        if (strpos($path, "/admin") !== 0) {
+            $path = "/admin/api/$apiVersion$path";
```

Comment:
instead of assuming `$path` has a `/` with the `getRequestPath` helper, can we instead:

```suggestion
        if ($path[0] != '/') {
            $path = "/admin/api/$apiVersion/$path";
```

Reply:
`parent::getRequestPath` is set up to always ensure the path has a `/` at position 0, which of course we could change here, but I wasn't too keen on using just the slash as a marker for a full path because I felt it would be too error prone.

My main concern was that folks would try things like `/product.json`, which I think should just be prefixed with the default.